src/ndisrcdemux/imp.rs

```diff
@@ -280,4 +287,26 @@ impl NdiSrcDemux {
         let mut state = self.state.lock().unwrap();
         state.combiner.update_pad_flow(&srcpad, res)
     }
+
+    fn sink_event(&self,
+        pad: &gst::Pad,
+        element: &super::NdiSrcDemux,
+        event: gst::Event
+    ) -> bool {
+        use gst::EventView;
+
+        gst_log!(CAT, obj: pad, "Handling event {:?}", event);
+        if let EventView::Eos(_) = event.view() {
+            if pad.mode() != gst::PadMode::Pull && element.num_src_pads() == 0 {
```

Comment:
This element never works in pull mode so you don't have to check for that.

Otherwise all good, thanks!